src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -426,11 +428,8 @@ impl VfioDevice {
             _ => None,
         }
     }
-}
 
-#[async_trait]
-impl Device for VfioDevice {
-    async fn attach(&mut self, h: &dyn hypervisor) -> Result<()> {
+    fn set_vfio_device(&mut self) -> Result<&Self> {
```

Comment:
I'm not quite clear as to the return value change, could you explain?  I don't see an `Ok()` return value in the function implementation other than `Ok(self)`, so the function always returns `self` (if it succeeds).  However, seeing as the function already takes `self` as a mutable reference, wouldn't it be more natural if it just mutated `self` as needed and returned `Result<()>` as it did so far?

Reply:
Thx @pmores, good point! I changed the set_vfio_device -> Result<&Self> to set_vfio_device -> Result<()>, which reduces the use of clone.